app/api/leaderboard/functions.ts

```diff
@@ -113,3 +114,46 @@ export default async function fetchGitHubReleases(
     )
     .slice(0, sliceLimit);
 }
+
+interface RepositoriesResponse {
+  organization: {
+    repositories: {
+      pageInfo: {
+        hasNextPage: boolean;
+        endCursor: string | null;
+      };
+      nodes: Repository[];
+    };
+  };
+}
+export async function fetchAllReposName() {
+  let repos: string[] = [];
+  let afterCursor: string | null = null;
+
+  do {
+    const response: RepositoriesResponse = await octokit.graphql({
+      query: `
+        query GetTop10ActiveRepos($org: String!, $after: String) {
+          organization(login: $org) {
+            repositories(first: 20, after: $after, orderBy: { field: UPDATED_AT, direction: DESC }) {
+              pageInfo {
+                hasNextPage
+                endCursor
+              }
+              nodes {
+                name
+              }
+            }
+          }
+        }
+      `,
+      org: env.NEXT_PUBLIC_GITHUB_ORG,
+      after: afterCursor,
+    });
+    const { nodes, pageInfo } = response.organization.repositories;
+    repos.push(...nodes.map((repo) => repo.name));
+    afterCursor = pageInfo.hasNextPage ? pageInfo.endCursor : null;
+  } while (afterCursor !== null);
```

Comment:
Repeating the same review again..., Instead of manually traversing through the pages, use `octokit.graphql.paginate` instead.

Refer: https://github.com/octokit/plugin-paginate-graphql.js/